connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceTask.java

```diff
@@ -315,18 +314,16 @@ static class PartitionState {
 
         // true if we should emit an offset sync
         boolean update(long upstreamOffset, long downstreamOffset) {
-            // This value is what OffsetSyncStore::translateOffsets would compute for this offset given the last sync.
-            // Because this method is called at most once for each upstream offset, simplify upstreamStep to 1.
+            // Emit an offset sync if any of the following conditions are true
+            boolean noPreviousSyncThisLifetime = lastSyncDownstreamOffset == -1L;
+            // the OffsetSync::translateDownstream method will translate this offset 1 past the last sync, so add 1.
             // TODO: share common implementation to enforce this relationship
-            long downstreamTargetOffset = lastSyncDownstreamOffset + 1;
-            if (lastSyncDownstreamOffset == -1L
-                    || downstreamOffset - downstreamTargetOffset >= maxOffsetLag
-                    || upstreamOffset - previousUpstreamOffset != 1L
```

Comment:
Why are we removing this condition?

Reply:
Oops this leaked from another fix, reverting.